### PR TITLE
Form linting updates

### PR DESCRIPTION
**Pull request by paulsturgess**
To prepare for the future upgrade of [carbon-factory](https://github.com/Sage/carbon-factory), that will introduce stricter linting rules, this PR updates the `form` components to pre-emptively resolve the errors.

#### Updated components
- [x] Checkbox
- [x] Fieldset
- [x] Filter (no changes required)
- [x] GroupedCharacter
- [x] RadioButton
- [x] Textarea
- [x] Textbox

#### Notes
- I've set Checkbox `checked` and `value` prop types as boolean. Not 100% sure that is correct as it feels like some developers might have set them as strings.

#### Components specifically not included
- Form
- FormSummary

Linting updates for these components have been included in commit https://github.com/Sage/carbon/pull/1243/commits/92600a999d9acf09444ca7583e97614621e0fc97 which is in a separate PR.

### CODE REVIEW

File: src/components/radio-button/radio-button.js [hunk 1]
```diff
@@ -30,14 +31,32 @@ import { tagComponent } from '../../utils/helpers/tags';
...
 const RadioButton = Input(InputLabel(InputValidation(
 class RadioButton extends React.Component {
+  static propTypes = {
+    /**
+     * A custom class name for the component.
+     *
+     * @property className
+     * @type {String}
+     */
+    className: PropTypes.string,
```
Comment (SeanArmstrong) [inline src/components/radio-button/radio-button.js]: I wonder due to the safeProps thing if defining className here will stop it being applied correctly. I don't think in general we need to be defining these kind of prop types that are always available

Comment (paulsturgess) [inline src/components/radio-button/radio-button.js]: It looks like it is possible to whitelist some props that will always be ignored: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prop-types.md

**Comment by toni-leigh** [comment]
I would revert the changes to `Form` and `FormSummary` in this PR as we have [this PR here](https://github.com/Sage/carbon/pull/1243) - I'm just finding out off @tomajwinter whether he's running linting on his PR

**Comment by paulsturgess** [comment]
@toni-leigh I've removed the two commits that updated `Form` and `FormSummary` as @tomajwinter has covered the changes in his PR.

**Comment by sianford** [comment]
Tested all components apart from InlineInputs in Safari, Chrome and Firefox browsers. Components behave as expected. Awaiting a demo component for InlineInputs to test.

**Comment by paulsturgess** [comment]
@SeanArmstrong I'm wondering if I should remove inlineInputs work from this PR as it's got some pretty big conflicts and I can see you've got another PR open to update the linting anyway... https://github.com/Sage/carbon/pull/1305

**Comment by SeanArmstrong** [comment]
@paulsturgess Yes sorry i meant to mention/message you about that

**Comment by paulsturgess** [comment]
I've removed the updates made to `inlineInputs`

**Comment by SeanArmstrong** [comment]
This has now passed QA as InlineInputs has been removed

**Pull request merged**